src/gemdat/plots/plotly/_shape.py

```diff
@@ -32,4 +34,131 @@ def shape(
     fig : plotly.graph_objects.Figure
         Output figure
     """
-    raise NotImplementedError
+    x_labels = ('X / Å', 'Y / Å', 'Z / Å')
+    y_labels = ('Y / Å', 'Z / Å', 'X / Å')
+
+    fig = make_subplots(
+        rows=2,
+        cols=3,
+        subplot_titles=x_labels,
+        shared_xaxes=True,
+        vertical_spacing=0.1,
+    )
+
+    distances = shape.distances()
+    distances_sq = distances**2
+
+    msd = np.mean(distances_sq)
+    std = np.std(distances_sq)
+    # Latex can not be rendered in 3D plotly plots: https://github.com/plotly/plotly.js/issues/608
+    title = f'{shape.name}: MSD = {msd:.3f} Å^2, std = {std:.3f}'
+
+    mean_dist = np.mean(distances)
+
+    _tmp_dict = defaultdict(list)
+    vector_dict = {}
+    if sites:
+        for site in sites:
+            _tmp_dict[site.label].append(site.coords)
+        for key, values in _tmp_dict.items():
+            vector_dict[key] = np.array(values) - shape.origin
+
+    coords = shape.coords
+
+    for col, (i, j) in enumerate(((0, 1), (1, 2), (2, 0))):
```

Comment:
Oof, this for loop is a bit of a beast 😅 I don't see an obvious way to make it better though.

Perhaps you can factor out the inner and outer loops like so, although that requires passing around a ton of variables:

```
    for col, (i, j) in enumerate(((0, 1), (1, 2), (2, 0))):
        _plot_contour(fig=fig, loc=top_row, coords=coords, ...)

        for label, vects in vector_dict.items():
            _plot_histo(fig=fig, loc=bot_row, vects=vects, ...)
```